src/homework/sandro/FourInARow/FourInARow.java

```diff
@@ -0,0 +1,305 @@
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i + 1][j] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i + 2][j] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i][j + 1] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i][j + 2] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (field[i][j + 3] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (count == 4) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+    public static boolean detectWinDiagonalfront1(int[][] field, int row, int col, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < 4; i++) {
+            for (int j = 0; j < 3; j++) {
+                if (field[i][j] == currentplayer
+                        || field[i + 1][j] == currentplayer
+                        || field[i + 2][j] == currentplayer
+                        || field[i][j + 1] == currentplayer
+                        || field[i][j + 2] == currentplayer
+                        || field[i][j + 3] == currentplayer) {
+                    count++;
+                } else {
+                    count = 0;
+                }
+                if (count == 4) {
+                    return true;
+                }
+            }
+        }
+        return false;
+    }
+
+    public static boolean detectWinDiagonalback(int[][] field, int currentplayer) {
+        int count = 0;
+        for (int i = 0; i < field.length; i++) {
+            if (field[i][field.length - i - 1] == currentplayer
+                    || field[i + 1][field.length - i - 1] == currentplayer
+                    || field[i + 2][field.length - i - 1] == currentplayer
+                    || field[i][field.length - (i + 1) - 1] == currentplayer
+                    || field[i][field.length - (i + 2) - 1] == currentplayer
+                    || field[i][field.length - (i + 3) - 1] == currentplayer) {
+                count++;
+            } else {
+                count = 0;
+            }
+            if (count == 4) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+
+    public static boolean detectWinRow(int[][] field, int row, int currentplayer) {         //do i need a int row (from "i") or could i just use field.length ?
+        int count = 0;
+        for (int i = 0; i < field[row].length; i++) {
```

Comment:
field[row].length stands for your col count while field.length stands for your row count. These values won't change. This is why I suggest that you replace them with two final static constants (ROW_COUNT, COL_COUNT). Makes it a lot more readable. You can then also replace the `6` and `7` that you have put all over the code (e.g. as a parameter for your printBoard method) with these two constants. Makes it so that if you want to change the values, you only have to change them at a single place.